### PR TITLE
[json-rpc] add content-length limit

### PR DESCRIPTION
**Pull request by xli**
Add content-length limit to json-rpc server.

### CODE REVIEW

File: json-rpc/src/tests/unit_tests.rs [hunk 1]
```diff
@@ -1318,3 +1326,11 @@ fn execute_batch_and_get_first_response(
         .remove(0)
         .unwrap()
 }
+
+fn gen_string(len: usize) -> String {
```
Comment (thefallentree) [inline json-rpc/src/tests/unit_tests.rs:1330]: https://docs.rs/rand/0.7.3/rand/trait.Rng.html#method.fill maybe faster? I don't actually know

Comment (xli) [reply json-rpc/src/tests/unit_tests.rs:1330]: not sure, but prefered Alphanumeric string, all random may produce double quote, which may break json format. so will just keep this version

**Pull request closed without merging**